### PR TITLE
feat(data-table): merge sticky header style

### PR DESCRIPTION
**Pull request by asudoh**
Merges @jnm2377's sticky header code.

#### Changelog

**New**

- Sticky header style (originally from @jnm2377).

**Changed**

- Fix missing overflow menu column CSS class in template.

#### Testing / Reviewing

Testing should make sure data table is not broken.

### CODE REVIEW

**Comment by asudoh** [comment]
Sticky header UI can be seen at https://deploy-preview-2965--the-carbon-components.netlify.com/component/data-table--default.html and https://deploy-preview-2965--the-carbon-components.netlify.com/component/data-table--with-pager.html.

File: packages/components/src/components/data-table/_data-table-core.scss [hunk 1]
```diff
@@ -425,6 +428,88 @@
...
+  // -------------
+  .#{$prefix}--data-table_inner-container {
+    background-color: $ui-03;
+    padding-top: rem(48px);
+    transform: translateZ(0);
+  }
+
+  .#{$prefix}--data-table--sticky-header {
+    display: block;
+    // max-height: rem(300px);
```
Comment (emyarod) [inline packages/components/src/components/data-table/_data-table-core.scss:443]:
```suggestion

```

**Review by emyarod — DISMISSED** [review]
looks good to me, we can probably remove the commented code

**Review by emyarod — COMMENTED** [review]
Kapture 2019-06-07 at 7 58 55

it looks like data table row height increases on hover now

**Review by dakahn — CHANGES_REQUESTED** [review]
Seeing the same bug as @emyarod 🐛

**Comment by asudoh** [comment]
@emyarod @dakahn Thank you for spotting the row height! Fixed.

**Review by emyarod — APPROVED** [review]
looks good to me!

**Comment by aagonzales** [comment]
In this deploy preview the pagination isn't aligning with the data table: 
https://deploy-preview-2965--the-carbon-components.netlify.com/component/data-table--with-pager.html

image

**Comment by asudoh** [comment]
Thanks you for spotting that @aagonzales! Fixed.

**Comment by aagonzales** [comment]
For the batch action version, can we keep the border-bottom on the column header? So that the selected backgrounds don't blend in with the column header background?

Before scrolling:
image

While scrolling (as-is):
image

**Comment by asudoh** [comment]
Good finding @aagonzales - Fixed.

**Review by dakahn — APPROVED** [review]

**Comment by joshblack** [comment]
cc @aagonzales if you have a chance to re-review 👍

**Review by shixiedesign — COMMENTED** [review]
Hi @asudoh, sorry I clearly remember reviewing this with gifs but somehow I'm not seeing my comment so here's my notes: 
- animation of the action bar appears broken: 
_Current (at 10% speed)_
Jun-17-2019 10-49-21
_Should be_
56475145-f823d980-6449-11e9-8a1c-2226af00c362

**Review by joshblack — CHANGES_REQUESTED** [review]
Mirroring @shixiedesign's comments so the review state is displayed correctly

**Comment by asudoh** [comment]
@shixiedesign Thank you for catching this - Seems that it happens in `master` (at https://the-carbon-components.netlify.com/component/data-table--select.html), so addressing it in a separate PR seems to make more sense.

**Review by shixiedesign — APPROVED** [review]

**Pull request merged**